libs/partners/anthropic/langchain_anthropic/chat_models.py

```diff
@@ -1114,27 +1068,76 @@ def _make_message_chunk_from_anthropic_event(
     event: anthropic.types.RawMessageStreamEvent,
     *,
     stream_usage: bool = True,
+    coerce_content_to_string: bool = False,
 ) -> Optional[AIMessageChunk]:
     """Convert Anthropic event to AIMessageChunk.
 
     Note that not all events will result in a message chunk. In these cases
     we return None.
     """
     message_chunk: Optional[AIMessageChunk] = None
+    # See https://github.com/anthropics/anthropic-sdk-python/blob/main/src/anthropic/lib/streaming/_messages.py  # noqa: E501
     if event.type == "message_start" and stream_usage:
         input_tokens = event.message.usage.input_tokens
         message_chunk = AIMessageChunk(
-            content="",
+            content="" if coerce_content_to_string else [],
             usage_metadata=UsageMetadata(
                 input_tokens=input_tokens,
                 output_tokens=0,
                 total_tokens=input_tokens,
             ),
         )
-    # See https://github.com/anthropics/anthropic-sdk-python/blob/main/src/anthropic/lib/streaming/_messages.py  # noqa: E501
-    elif event.type == "content_block_delta" and event.delta.type == "text_delta":
-        text = event.delta.text
-        message_chunk = AIMessageChunk(content=text)
+    elif (
+        event.type == "content_block_start"
+        and event.content_block is not None
+        and event.content_block.type == "tool_use"
+    ):
+        if coerce_content_to_string:
+            warnings.warn("Received unexpected tool content block.")
+        # TODO: Pass type through here
```

Comment:
should we make all the keys line up here (and store a `content_block` field), or better to re-form?

Reply:
Purely because of `type` string fields - right now they would concatenate repeatedly. Could change in core but am wary of magic params.

Reply:
I think we will need the type field to be able to send these messages back to Anthropic (we also [raise ValueError](https://github.com/langchain-ai/langchain/blob/c1fced926923f7cf9bb982e3d0c610aa9dff6f35/libs/partners/anthropic/langchain_anthropic/chat_models.py#L179) without them).

I'm also on board to add `type` as another magic param in core. The risk would be if providers stream out values from the `"type"` key.

Not 100% on this but another idea is to keep track of a mapping from `index` to `type` within ChatAnthropic and pop out the type key from all chunks after the first one per value of `index`.